ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activityAgreementAnswer.component.spec.ts

```diff
@@ -85,10 +85,10 @@ describe('ActivityAgreementAnswer', () => {
         const checkbox = debugElement.queryAll(By.css('mat-checkbox'))[0];
         const inputElement = checkbox.nativeElement.querySelector('input') as HTMLInputElement;
         inputElement.click();
-        fixture.detectChanges();
         expect(inputElement.checked).toBe(true);
+        expect(component.block.answer).toBe(true);
         inputElement.click();
-        fixture.detectChanges();
         expect(inputElement.checked).toBe(false);
+        expect(component.block.answer).toBe(false);
```

Comment:
For some reason test was a bit unstable due to these detectChanges.
In 2 out of the 8 cases test was failed.\
Also added additional expects.

Reply:
Exactly when and how change detection occurs in tests is something I continue to be confused about!